lib/syskit/roby_app/rest_api.rb

```diff
@@ -0,0 +1,181 @@
+            rescue_from RESTDeploymentManager::NotFound do |e|
+                error! e.message, 404
+            end
+
+            rescue_from RESTDeploymentManager::Forbidden do |e|
+                error! e.message, 403
+            end
+
+            helpers Roby::Interface::REST::Helpers
+            helpers do
+                def syskit_conf
+                    Syskit.conf
+                end
+
+                # The {DeploymentManager} object that manages the deployments on behalf of the API
+                def deployment_manager
+                    @contexts ||=
+                        (roby_storage['syskit.deployment_manager'] ||= RESTDeploymentManager.new(syskit_conf))
+                end
+
+                # @api private
+                #
+                # Transforms orogen's deployed task information object into the
+                # hash that is to be returned by the API
+                def make_basic_deployed_task_info_hash(info)
+                    info_hash = info.to_h
+                    if /^orogen_default_/.match?(info.task_name)
+                        info_hash[:default_deployment] = true
+                    end
+                    if info.task_model_name == 'logger::Logger' && (info.task_name == info.deployment_name + "_Logger")
+                        info_hash[:default_logger] = true
+                    end
+                    info_hash
+                end
+
+                def make_configured_deployment_info(d)
+                    type = find_type_from_process_server(
+                        syskit_conf.process_server_config_for(d.process_server_name))
+                    return if !type
+
+                    tasks = d.each_orogen_deployed_task_context_model.map do |deployed_task|
+                        Hash['task_name' => deployed_task.name,
+                             'task_model_name' => deployed_task.task_model.name]
+                    end
+
+                    Hash[id: d.object_id,
+                         created: deployment_manager.created_here?(d),
+                         deployment_name: d.process_name,
+                         tasks: tasks,
+                         on: d.process_server_name,
+                         mappings: d.name_mappings,
+                         type: type]
+                end
+
+                PROCESS_SERVER_TYPES = Hash[
+                    Orocos::RemoteProcesses::Client => 'orocos',
+                    UnmanagedTasksManager => 'unmanaged'
+                ]
+
+                # Returns a string that describes the process server type
+                #
+                # @return [String,nil] the description, or nil if the given
+                #   process server should not be exposed by the REST API
+                def find_type_from_process_server(process_server_config)
+                    PROCESS_SERVER_TYPES[process_server_config.client.class]
+                end
+            end
+
+            get '/deployments/available' do
```

Comment:
These APIs usually provide some kind of filtering.. IMO, filter deployments/tasks within a given project and deployments/tasks that contain task models defined in a given project would be very useful

Reply:
There will never be that many available deployments. My policy there would be to filter at the client level.

Filtering in "real web apps" is usually present because (1) there can be a large number of responses and (2) it is easier on the database (gain when there's a lot of clients)